File: leetcode/problems/language-specific/go/191.number-of-1-bits.go
/*
 * @lc app=leetcode id=191 lang=golang
 *
 * [191] Number of 1 Bits
 */

// @lc code=start

// * Bit Manipulation Solution | O(n) Time | O(1) Space

func hammingWeight(num uint32) int {
	count := uint32(0)
	for i := 0; i < 32; i++ {
		count += num & 1
		num >>= 1
	}

	return int(count)
}

// @lc code=end

File: leetcode/problems/sessions/2025-10-25_2025-??-??/191.number-of-1-bits.py
#
# @lc app=leetcode id=191 lang=python3
#
# [191] Number of 1 Bits
#

# @lc code=start

# * Bit Manipulation Solution | O(logn) Time | O(1) Space
# * n -> The given input


class Solution:
    def hammingWeight(self, n: int) -> int:
        res = 0
        while n:
            res += n & 1
            n >>= 1
        return res


# @lc code=end
